src/Test/Hspec/AnnotatedException.hs

```
@@ -0,0 +1,111 @@
+module Test.Hspec.AnnotatedException
+  ( unwrapAnnotatedHUnitFailure
+  , annotateHUnitFailure
+  ) where
+
+import Prelude
+
+import Control.Exception (Exception)
+import Control.Exception qualified
+import Control.Exception.Annotated (AnnotatedException (..))
+import Control.Lens (Lens', lens, over)
+import Data.Annotation (Annotation, tryAnnotations)
+import Data.Foldable (toList)
+import Data.List.NonEmpty (NonEmpty ((:|)), nonEmpty)
+import Data.Maybe (catMaybes, listToMaybe)
+import Data.Text (Text)
+import Data.Text qualified as T
+import GHC.Stack (CallStack, prettyCallStack)
+import Test.HUnit.Lang (FailureReason (..), HUnitFailure (..))
+import Test.Hspec
+
+-- | An hspec hook that lets hspec catch and pretty-print 'HUnitFailure', the
+--   exception that is thrown when a test assertion fails
+--
+-- Tests for any code that might throw 'AnnotatedException' (which includes anything
+-- that uses freckle-app) should add this hook to their test suite. Without it, if
+-- you end up with an @'AnnotatedException' 'HUnitFailure'@, hspec doesn't recognize
+-- it as an assertion failure and you get ugly output instead of nice output.
```

Can we confirm here that it is _still_ a failure, even if it is not an 'assertion' failure? In other words, the exception will not be swallowed up and therefore have a test that should have failed pass. It might seem obvious now but it's important to get clarity here IMHO.